src/README.md

```diff
@@ -101,6 +101,44 @@ There is also a file watcher, which monitors the `content` directory and automat
 npm run watch
 ```
 
+## Generating chapter images
+
+We can automate the generation of chapter images from the command line to save this onerous task.
+
+It can be run like below, by passing a chapter markdown (with or without the `.md` extension):
+
+```
+npm run figure-images en/2021/pwa
+```
+
+Which will then generate any missing figures based on the chapter markup, skipping images that already exist:
+
+```
+> almanac.httparchive.org@0.0.1 figure-images
+> node ./tools/generate/generate_figure_images "en/2021/pwa"
+
+Generating for chapter: pwa for year 2021
+  Skipping: pwa-service-worker-controlled-pages-by-rank.png as image already exists
+  Skipping: pwa-most-used-service-worker-events.png as image already exists
+  Skipping: pwa-service-worker-and-manifest-usage.png as image already exists
+  Skipping: pwa-top-pwa-manifest-properties.png as image already exists
+  Skipping: pwa-top-pwa-manifest-icon-sizes.png as image already exists
+  Skipping: pwa-manifest-display-values.png as image already exists
+  Skipping: pwa-manifests-preferring-native-app.png as image already exists
+  Skipping: pwa-industry-categories.png as image already exists
+  Skipping: pwa-lighthouse-pwa-audits.png as image already exists
+  Skipping: pwa-lighthouse-pwa-scores.png as image already exists
+  Skipping: pwa-libraries-and-scripts.png as image already exists
+  Skipping: pwa-top-workbox-versions.png as image already exists
+  Skipping: pwa-top-workbox-packages.png as image already exists
+  Generating image pwa-workbox-runtime-caching-strategies.png...
+  Generating image pwa-notification-acceptance-rates.png...
+  Generating image pwa-install-events.png...
+```
+
+Authors can delete images and rerun if they want to, to regenerate images.
```

Comment:
Have you considered a `--force` flag that overwrites any existing images?

Reply:
I did.

There's pluses and minus's:

-ve:
- More complexity. Likely need a new dependency. Looked at a few but decided to submit what we have now. As mentioned can just delete the files for now.
- Risk of over deleting when just changing one file. And as images will be optimised (by Tinypng or Calibre) will just be more Git noise (though did discover that Tinypng does do it consistently so updated all images and only the changed ones were recognised as changed).

+ve:
- I've deleted non-chart images when regenerating the images for resource-hints. Easy enough to git checkout again, but still, an `-f` flag would avoid this.

I think we should merge an iterate.